docs/smart-contracts/guides/conventions/work-contractspec-js.mdx

```diff
@@ -4,4 +4,551 @@ hide_table_of_contents: true
+
+#### Step-by-Step Guide
+
+##### Example Contract: TokenManager
+
+Here's an example Rust contract that we'll be working with. This contract, `TokenManager`, manages token balances and includes functions to initialize, deposit, withdraw, and check balances. It also includes custom errors for different failure scenarios.
+
+```rust
+use soroban_sdk::{contract, contractimpl, Address, Env, Symbol, contracterror};
+
+#[contract]
+pub struct TokenManager;
+
+#[contracterror]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum TokenManagerError {
+    InternalError = 1,
+    UnauthorizedError = 2,
+    InsufficientBalanceError = 3,
+    OverflowError = 4,
+}
+
+#[contractimpl]
+impl TokenManager {
+    pub fn initialize(e: Env, admin: Address) {
+        e.storage().set(&Symbol::short("admin"), &admin);
+    }
+
+    pub fn deposit(e: Env, user: Address, amount: i128) -> Result<(), TokenManagerError> {
+        let key = Symbol::short("balance").join(&user);
+        let mut balance: i128 = e.storage().get(&key).unwrap_or(0);
+        balance = balance.checked_add(amount).ok_or(TokenManagerError::OverflowError)?;
+        e.storage().set(&key, &balance);
+        Ok(())
+    }
+
+    pub fn withdraw(e: Env, user: Address, amount: i128) -> Result<(), TokenManagerError> {
+        let key = Symbol::short("balance").join(&user);
+        let mut balance: i128 = e.storage().get(&key).unwrap_or(0);
+        if balance < amount {
+            return Err(TokenManagerError::InsufficientBalanceError);
+        }
+        balance -= amount;
+        e.storage().set(&key, &balance);
+        Ok(())
+    }
+
+    pub fn get_balance(e: Env, user: Address) -> i128 {
+        let key = Symbol::short("balance").join(&user);
+        e.storage().get(&key).unwrap_or(0)
+    }
+}
+```
+
+##### Step 1: Deploy the Contract
+
+First, deploy your Soroban smart contract to the network.
+
+```bash
+soroban contract deploy --source orbit --network testnet --wasm token_manager.wasm
+```
+
+This command will return the contract ID, for example: `CBEUVPRZKMCX5MPUAF2JMKMRIFEXXJXQXBLUUW6LMN5GDDQQSX3UHYXA`.
+
+##### Step 2: Generate TypeScript Bindings
+
+Use the Soroban CLI to generate the TypeScript bindings which include the contract specification.
+
+```bash
+soroban contract bindings typescript --output-dir testbindings --network testnet --contract-id CBEUVPRZKMCX5MPUAF2JMKMRIFEXXJXQXBLUUW6LMN5GDDQQSX3UHYXA
+```
+
+This command will create TypeScript bindings in the specified directory.
+
+Alternatively, you can get your spec using javascript: credit to [@mason in Stellar Developers discord](https://discord.com/channels/897514728459468821/966788672164855829/1201708320336138271) and [his gist here] (https://gist.github.com/masonforest/27ebfafa3452227937b842c56731d817)
```

Comment:
nit:
```suggestion
Alternatively, you can get your spec using pure JavaScript: credit to [@mason in Stellar Developers discord](https://discord.com/channels/897514728459468821/966788672164855829/1201708320336138271) and [his gist here] (https://gist.github.com/masonforest/27ebfafa3452227937b842c56731d817)
```

Reply:
Also this can probably be updated to use the new `rpc.Server.getContractWasm` et al. methods introduced in https://github.com/stellar/js-stellar-sdk/pull/960

Reply:
Since clients won't always have the WASM available.

Reply:
This could probably be a `Step 2a` or `Step 2 (Bonus)` section to separate it from the main method.

Reply:
^ still applies